.circleci/test-deploy.yml

```diff
@@ -15,8 +15,6 @@ jobs:
         - image: cimg/base:current
       steps:
         - checkout
-        # Run your orb's commands to validate them.
-        - dast/greet
```

Comment:
Im not seeing an example in here that uses env variables to pass secrets (I mean project env variables https://circleci.com/docs/2.0/env-vars/#setting-an-environment-variable-in-a-project).

I've tried to set up the test in a lot of different ways to use my already defined env variables but they're just not beign replaced from what I had setup on them. Is there something I am missing? As for now I've left them as `${MY_ENV_NAME}` on my `test-deploy.yml` file

Reply:
Try `$MY_ENV_NAME` instead of `${MY_ENV_NAME}`.

If it still doesn't work, you can try using a [context](https://circleci.com/docs/2.0/contexts/#creating-and-using-a-context) instead. If you do so, you should reference the context in your job like this:

```
- command-tests:
    filters: *filters
    context: my-context
```

This should to the trick.

Reply:
I've tried with those two suggestions but none of them seem's to work on this case (see latest two commits), It's a bit weird becuase I've noticed that in both runs in the `Preparing env variables` step on the CircleCI it seems that both of my env variables gets mentioned in the logs but then they don't get the value passed

Reply:
Can you add a `set -x` before the line below:

https://github.com/soos-io/soos-dast-circleci-orb/pull/1/files#diff-fc755159075e6d0769d6d8357e8dd27d9cd097673c741948fc057f5ac53bb589R55

This way, we will be able to see what is sent on the command line.

Reply:
Just added that line

Reply:
Thanks!

I'm looking at the pipeline, and the python command is receiving a literal `$SOOS_CLIENT_ID` and `$SOOS_API_KEY` as arguments.

You would have to expand the variables before using them. My suggestion here is to change the type of both parameters to [env_var_name](https://circleci.com/docs/2.0/reusing-config/#environment-variable-name) in the command and job definition. And then, in your script, expand the variables like this:

```
PARAMS="--clientId ${!SOOS_CLIENT_ID} --apiKey ${!SOOS_API_KEY} --projectName ${SOOS_PROJECT_NAME} --scanMode ${SOOS_SCAN_MODE} --apiURL ${SOOS_API_BASE_URL} --integrationName ${SOOS_INTEGRATION_NAME} --commitHash ${CIRCLE_SHA1} --branchName ${CIRCLE_BRANCH}"
```

Bear in mind that with this strategy [$SOOS_CLIENT_ID](https://github.com/soos-io/soos-dast-circleci-orb/blob/3ad7caf94d25984986c55f20f43db90b25693178/.circleci/test-deploy.yml#L19) won't be a valid parameter anymore. You must lose the dollar sign: `SOOS_CLIENT_ID`.

Reply:
tested with that approach but it doesn't seem to work either :confused: (last commit, also did apply the -eq 1 where it was missing)

Reply:
Thanks for doing the boolean update!

I believe it's not working now because the context variable has the same name as the variable you are using to hold it. This causes the environment variable value to be overwritten.

Try changing the [command](https://github.com/soos-io/soos-dast-circleci-orb/pull/1/files#diff-3a03d56ca00b8b5c9605f24d9ced1b12d8adb65cbcec2c90c9d7b184465c3936R99) environment parameters to `SOOS_CLIENT_ID_VAR_NAME` and `SOOS_API_KEY_VAR_NAME`. Make sure to change them in the Python command too.

Reply:
@EricRibeiro Thank you!
Seems that finally that one did the trick!